src/index.js

```diff
@@ -1,41 +1,194 @@
 'use strict'
 
 const Swarm = require('libp2p-swarm')
-const Peer = require('peer-info')
+const PeerId = require('peer-id')
+const PeerInfo = require('peer-info')
+const PeerBook = require('peer-book')
 const TCP = require('libp2p-tcp')
 // const UTP = require('libp2p-utp')
 const WS = require('libp2p-websockets')
 const spdy = require('libp2p-spdy')
 const multiaddr = require('multiaddr')
+const mafmt = require('mafmt')
 const EE = require('events').EventEmitter
 
 exports = module.exports
 
-exports.Node = function Node (peerInfo) {
+const OFFLINE_ERROR = new Error('The libp2p node is not started yet')
```

Comment:
Not good, this will make the stack trace come always from this line, rather then from where it's thrown. See this stackoverflow answer: http://stackoverflow.com/a/32749533 for a very good way to extend the `Error` object to a custom error. and then `throw` in the right place.